### PR TITLE
Feature bulk load

### PR DESCRIPTION
**Pull request by digitalBush**

### CODE REVIEW

File: spec/integration/bulkLoad.spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+const { config } = testHelpers;
+
+const skwell = require( "src/" );
```
Comment (calvinb) [inline spec/integration/bulkLoad.spec.js]: This works, but the trailing slash isn't idiomatic, is it?

File: spec/integration/bulkLoad.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+describe( "Bulk Load - Integration", () => {
+	let sql;
+	before( async () => {
+		sql = await skwell.connect( config );
+		return sql.execute( sql.fromFile( "sql/bulkLoad-setup.sql" ) );
+	} );
+
+	afterEach( async () => {
+		await sql.execute( "DELETE FROM BulkLoadTest" );
+		await sql.execute( "DROP TABLE IF EXISTS BulkLoadTestNew" );
```
Comment (calvinb) [inline spec/integration/bulkLoad.spec.js]: I think you could combine these and eliminate a round trip.

File: spec/integration/bulkLoad.spec.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+				rows: [ { id: 1 } ]
+			} ).should.eventually.be.rejectedWith( Error, "Unable to load temp table '#bulkloadnope' using connection pool. Use a transaction instead." );
+		} );
+
+		it( "should bulk load a table", async () => {
+			await sql.bulkLoad( "BulkLoadTest", {
+				schema: {
+					id: sql.int.nullable()
+				},
+				rows: [ { id: 1 }, { id: 2 }, { id: 3 } ]
```
Comment (calvinb) [inline spec/integration/bulkLoad.spec.js:39]: Since `id` is nullable, how about adding a null one?

File: spec/integration/bulkLoad.spec.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+			} );
+			const rows = await sql.query( "select * from BulkLoadTest" );
+			rows.should.deep.equal( [ { id: 1 }, { id: 2 }, { id: 3 } ] );
+		} );
+
+		it( "should create and bulk load a table", async () => {
+			await sql.bulkLoad( "BulkLoadTestNew", {
+				create: true,
```
Comment (calvinb) [inline spec/integration/bulkLoad.spec.js:47]: I like this

File: src/TypeWrapper.js [hunk 5]
```diff
@@ -9,4 +9,9 @@ module.exports = class TypeWrapper {
 		}
 	}
 
+	nullable( val = true ) {
+		this.isNull = val;
+		return this;
+	}
```
Comment (calvinb) [inline src/TypeWrapper.js:15]: ❤️

File: spec/integration/bulkLoad.spec.js (hunk 3 above)
Comment (digitalBush) [reply spec/integration/bulkLoad.spec.js:39]: Added a similar test below this one.

**Comment by digitalBush** [comment]
@calvinb thanks for the suggestions!

File: src/client.js [hunk 6]
```diff
@@ -40,6 +40,15 @@ class Client extends Api {
 		} );
 	}
 
+	// Override base API to keep from loading a temp table on different connection.
+	async bulkLoad( tableName, options ) {
+		if ( tableName.indexOf( "#" ) >= 0 ) {
```
Comment (calvinb) [inline src/client.js]: Sorry I just noticed this. I believe a permanent table can have a `#` in its name, so this should be `===`.

Comment (digitalBush) [reply src/client.js]: @calvinb cool, I'll look into this. I bet I need to split on dots too?

Comment (calvinb) [reply src/client.js]: Good idea.

Comment (digitalBush) [reply src/client.js]: Fixed! Let me know how that looks.

**Pull request merged**